scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -203,14 +203,13 @@ sealed trait Execution[+T] extends java.io.Serializable { self: Product =>
   override val hashCode: Int = ScalaRunTime._hashCode(self)
 
   override def equals(other: Any): Boolean = {
-    //removed from Scala 2.12
-    def _equals(x: Product, y: Any): Boolean = y match {
-      case y: Product if x.productArity == y.productArity =>
-        x.productIterator sameElements y.productIterator
+    val productEquals = other match {
+      case p: Product if self.productArity == p.productArity =>
```

Comment:
Updated this. Not sure how large those products get. I moved the hashCode into the case body.